pkg/client/cache/index.go

```diff
@@ -24,7 +24,11 @@ func ImageStreamReferenceIndexFunc(obj interface{}) ([]string, error) {
 			}
 			params := trigger.ImageChangeParams
 			name, _, _ := imageapi.SplitImageStreamTag(params.From.Name)
-			keys = append(keys, params.From.Namespace+"/"+name)
+			namespace := params.From.Namespace
```

Comment:
Shouldn't that happen in defaulting? If so, it'll happen client-side in the controller if a namespaceless ICT is read from the API, right?